File: java/base/src/main/vsdk/toolkit/common/color/ColorRgba.java
package vsdk.toolkit.common.color;

public final class ColorRgba
{
    private double r;
    private double g;
    private double b;
    private double a;

    public ColorRgba()
    {
        this(0.0, 0.0, 0.0, 0.0);
    }

    public ColorRgba(double r, double g, double b, double a)
    {
        this.r = r;
        this.g = g;
        this.b = b;
        this.a = a;
    }

    public ColorRgba(ColorRgba other)
    {
        this(other.r, other.g, other.b, other.a);
    }

    public double getR() { return r; }
    public double getG() { return g; }
    public double getB() { return b; }
    public double getA() { return a; }

    public void setR(double v) { r = v; }
    public void setG(double v) { g = v; }
    public void setB(double v) { b = v; }
    public void setA(double v) { a = v; }

    public void set(ColorRgba other)
    {
        r = other.r;
        g = other.g;
        b = other.b;
        a = other.a;
    }
}

File: typescript/base/src/main/vsdk/toolkit/common/color/ColorRgba.ts
export class ColorRgba {
    public constructor(
        private r = 0,
        private g = 0,
        private b = 0,
        private a = 0,
    ) {}
    public getR(): number {
        return this.r;
    }
    public getG(): number {
        return this.g;
    }
    public getB(): number {
        return this.b;
    }
    public getA(): number {
        return this.a;
    }
    public setR(v: number): void {
        this.r = v;
    }
    public setG(v: number): void {
        this.g = v;
    }
    public setB(v: number): void {
        this.b = v;
    }
    public setA(v: number): void {
        this.a = v;
    }
    public set(o: ColorRgba): void {
        this.r = o.r;
        this.g = o.g;
        this.b = o.b;
        this.a = o.a;
    }
}
